.github/workflows/pr-check-pipfile.yml

```
@@ -0,0 +1,140 @@
+      # pushしたブランチでPRを作る
+      - name: Create PullRequest
+        uses: actions/github-script@v3
+        if: github.event.pull_request.head.repo.full_name == github.repository && steps.diff.outcome != ''
+        env:
+          PYTHON_VERSION: ${{ matrix.python-version }}
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            const common_params = {
+              owner: context.repo.owner,
+              repo: context.repo.repo
+            }
+            const pull_params = {
+              head: "refs/heads/pipfile-" + process.env.PYTHON_VERSION.replace('.', '_') + "-${{github.event.pull_request.head.ref}}",
+              base: "${{github.event.pull_request.head.ref}}",
+              ...common_params
+            }
+            const pulls_list_params = {
+              state: "open",
+              ...pull_params
+            }
+            console.log("call pulls.list:")
+            console.log(pulls_list_params)
+            github.pulls.list(pulls_list_params).then(list_res => {
+              if (list_res.data.filter(data => data.user.id === 41898282).length === 0) {
+                const pulls_create_params = {
+                  title: "pipfileが更新されたので直してあげたよ！PRをマージしてね！ #${{github.event.pull_request.number}}",
+                  body: "鳩の唐揚げおいしい！😋😋😋 #${{github.event.pull_request.number}}",
+                  ...pull_params
+                }
+                console.log("call pulls.create:")
+                console.log(pulls_create_params)
+                github.pulls.create(pulls_create_params).then(create_res => {
+                  const issues_add_assignees_params = {
+                    issue_number: create_res.data.number,
+                    assignees: ["${{github.event.pull_request.user.login}}"],
+                    ...common_params
+                  }
+                  console.log("call issues.addAssignees:")
+                  console.log(issues_add_assignees_params)
+                  github.issues.addAssignees(issues_add_assignees_params)
+                })
+              }
+            })
+      # 既にPipfile修正のPRがある状態で、手動でPipfileを修正した場合、Pipfile修正のPRを閉じる
+      - name: Close PullRequest
+        uses: actions/github-script@v3
+        if: github.event.pull_request.head.repo.full_name == github.repository && steps.diff.outcome == ''
+        env:
+          PYTHON_VERSION: ${{ matrix.python-version }}
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            const ref = "heads/pipfile-" + process.env.PYTHON_VERSION.replace('.', '_') + "-${{github.event.pull_request.head.ref}}"
+            const common_params = {
+              owner: context.repo.owner,
+              repo: context.repo.repo
+            }
+            const pulls_list_params = {
+              head: "refs/" + ref,
```

```suggestion
              head: "${{github.event.pull_request.head.repo.owner.login}}:pipfile-" + process.env.PYTHON_VERSION.replace('.', '_') + "-${{github.event.pull_request.head.ref}}",
```

このように指定するとうまく対象を絞れる
ref: https://developer.github.com/v3/pulls/#list-pull-requests